src/org/sosy_lab/java_smt/test/TokenizerTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of JavaSMT,
+ * an API wrapper for a collection of SMT solvers:
+ * https://github.com/sosy-lab/java-smt
+ *
+ * SPDX-FileCopyrightText: 2024 Dirk Beyer <https://www.sosy-lab.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.sosy_lab.java_smt.test;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.truth.Truth;
+import org.junit.Test;
+import org.sosy_lab.java_smt.basicimpl.Tokenizer;
+
+public class TokenizerTest {
+  @Test
+  public void validBrackets() {
```

Comment:
Writing that many test function is nice and show that you have thought bout several cases.
It would be simpler and better readable to combine several cases into one function.
For example:

```
public void testParserForValidInput() {
    assertThat(Tokenizer.tokenize("(assert (= 3 (+ 2 1)))")).containsExactly("(assert (= 3 (+ 2 1)))");
    assertThat(Tokenizer.tokenize("(assert (= 3;)\n(- 4 1)))")).containsExactly("(assert (= 3;)\n(- 4 1)))");
    ...
```

and 

```
public void testParserForInalidInput() {
    assertThrows(IllegalArgumentException.class, () -> Tokenizer.tokenize("(assert (= 3 (+ 2 1))))"));
    assertThrows(...);
    ...
```